src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/AwsStorageOptionsConfiguration.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using Fluid;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.FileStorage.AmazonS3;
+
+namespace OrchardCore.Media.AmazonS3;
+
+public class AwsStorageOptionsConfiguration : IConfigureOptions<AwsStorageOptions>
+{
+    private readonly IShellConfiguration _shellConfiguration;
+    private readonly ShellSettings _shellSettings;
+    private readonly ILogger _logger;
+
+    // Local instance since it can be discarded once the startup is over
+    private readonly FluidParser _fluidParser = new ();
+
+    public AwsStorageOptionsConfiguration(
+        IShellConfiguration shellConfiguration,
+        ShellSettings shellSettings,
+        ILogger<AwsStorageOptions> logger)
+    {
+        _shellConfiguration = shellConfiguration;
+        _shellSettings = shellSettings;
+        _logger = logger;
+    }
+
+    public void Configure(AwsStorageOptions options)
+    {
+        options.BindConfiguration(_shellConfiguration);
+
+        var templateOptions = new TemplateOptions();
+        var templateContext = new TemplateContext(templateOptions);
+        templateOptions.MemberAccessStrategy.Register<ShellSettings>();
+        templateOptions.MemberAccessStrategy.Register<AwsStorageOptions>();
+        templateContext.SetValue("ShellSettings", _shellSettings);
+
+        ParseBucketName(options, templateContext);
+        ParseBasePath(options, templateContext);
+    }
+
+    private void ParseBucketName(AwsStorageOptions options, TemplateContext templateContext)
+    {
+        // Use Fluid directly as this is transient and cannot invoke _liquidTemplateManager.
+        try
+        {
+            var template = _fluidParser.Parse(options.BucketName);
+
+            options.BucketName = template
+                .Render(templateContext, NullEncoder.Default)
+                .Replace("\r", String.Empty).Replace("\n", String.Empty);
```

Comment:
Why not just trim. If there are space in the middle of the string they should be intentional. I could understand that having a new line in a fluid template might be hardly noticeable though.

Reply:
Can we use `Environment.NewLine` instead?

Reply:
Instead of "\n"? So result will be smth like :
`            options.BucketName = template
                .Render(templateContext, NullEncoder.Default)
                .Replace("\r", String.Empty)
                .Replace(System.Environment.NewLine, String.Empty)
                .Trim();`

Reply:
`Environment.NewLine` insstead of `\r\n`

Reply:
```csharp
options.BucketName = template .Render(templateContext, NullEncoder.Default) .Replace(Environment.NewLine, String.Empty)
```

Reply:
```csharp
options.BucketName = template .Render(templateContext, NullEncoder.Default) .Replace(Environment.NewLine, String.Empty)
```

Reply:
Added, I've also added Trim() as Sebastien mention

Reply:
@hishamco

Yes `Environment.NewLine` is useful to generate strings on a given platform.

But not sure it is a good idea to cleanup strings that may come from different sources.

I would have let the replace of `\r` and `\n`.

Reply:
So guys, should i rollback my latest commit? :)

Reply:
I think it is safer, but keep the trim

Reply:
Done.

Reply:
> I would have let the replace of \r and \n.

Is there anything wrong with `Environment.NewLine`?